src/utils/queue.ts

```diff
@@ -0,0 +1,44 @@
+export interface QueueItem<T = any> {
+    fn: () => Promise<T>;
+    resolve: (value: T) => void;
+    reject: (reason?: any) => void;
+}
+
+let intervalId: NodeJS.Timeout | undefined = undefined;
+let firstItemInQueue: boolean = true;
+const requestQueue: QueueItem[] = [];
+
+export async function startProcessingQueue() {
+    if (intervalId) {
+        return;
+    }
+    intervalId = setInterval(async () => {
+        if (requestQueue.length > 0) {
+            const { fn, resolve, reject } = requestQueue.shift() as QueueItem;
+            try {
+                const result = await fn();
+                resolve(result);
+            } catch (error) {
+                reject(error);
+            }
+        }
+    }, 1000);
+}
+
+export function stopProcessingQueue() {
+  if (intervalId) {
+    clearInterval(intervalId);
+    intervalId = undefined;
+  }
+}
+
+export async function enqueueMethod<T>(fn: () => Promise<T>): Promise<T> {
+    if (firstItemInQueue) {
```

Comment:
this needs some more thought because subsequent delayed (> 1s) first enqueue calls will not get in this block while they should.

this variable needs to be reset to true after the last item has left the queue after 1s.